core/src/main/java/lucee/runtime/type/KeyImpl.java

```diff
@@ -192,7 +192,8 @@ public boolean equals(Object other) {
 			return key.equalsIgnoreCase((String) other);
 		}
 		if (other instanceof Key) {
-			return ucKey.equalsIgnoreCase(((Key) other).getUpperString());
+			// Both strings are guaranteed to be upper case
+			return ucKey.equals(((Key) other).getUpperString());
```

Comment:
as far as i remember I have choosen "equalsIgnoreCase" because it is faster, because "equalsIgnoreCase" argument is a string and the one from "equals" is an object. But we need to test this, maybe it also changed with never JVM.

Reply:
@michaeloffner No, a case-insensitive comparison should always be faster.  I found this bottleneck while profiling the code during my performance testing and it was internally changing the case of the strings.  Both objects being compared are a string.  The `ucKey` variable is already a string and `other.getUpperString()` also returns a string.  They are both guaranteed to be upper case so a case-insensitive comparison will be the fastest.

Reply:
agree i did some testing and the result was clear (i only test with different instances of the string)
`	public static void main(String[] args) {
		int max = 1000000;
		int loops = 10;
		String strEQ1 = "bjguguhbjbjgftzfzvhvbkjbkjbjkbj";
		String strEQ2 = new String(strEQ1.getBytes(CharsetUtil.UTF8), CharsetUtil.UTF8); // make sure we do not have the same instance
		String strOD = "bjguguhbjbjgftzfzvhvbkjbkjbjkbi";

		// equal true
		double fastestEqualTrue = Double.MAX_VALUE;
		for (int y = 0; y < loops; y++) {
			double start = SystemUtil.millis();
			boolean b;
			for (int i = 0; i < max; i++) {
				b = strEQ1.equals(strEQ2);
			}
			double tmp = SystemUtil.millis() - start;
			if (fastestEqualTrue > tmp) fastestEqualTrue = tmp;
		}
		print.e("equal true:" + fastestEqualTrue);

		// equal true
		double fastestEqualFalse = Long.MAX_VALUE;
		for (int y = 0; y < loops; y++) {
			double start = SystemUtil.millis();
			boolean b;
			for (int i = 0; i < max; i++) {
				b = strEQ1.equals(strOD);
			}
			double tmp = SystemUtil.millis() - start;
			if (fastestEqualFalse > tmp) fastestEqualFalse = tmp;
		}
		print.e("equal false:" + fastestEqualFalse);

		// equalsIgnoreCase true
		double fastestEqualIgnoreCaseTrue = Long.MAX_VALUE;
		for (int y = 0; y < loops; y++) {
			double start = SystemUtil.millis();
			boolean b;
			for (int i = 0; i < max; i++) {
				b = strEQ1.equalsIgnoreCase(strEQ2);
			}
			double tmp = SystemUtil.millis() - start;
			if (fastestEqualIgnoreCaseTrue > tmp) fastestEqualIgnoreCaseTrue = tmp;
		}
		print.e("equalsIgnoreCase true:" + fastestEqualIgnoreCaseTrue);

		// equalsIgnoreCase false
		double fastestEqualIgnoreCaseFalse = Long.MAX_VALUE;
		for (int y = 0; y < loops; y++) {
			double start = SystemUtil.millis();
			boolean b;
			for (int i = 0; i < max; i++) {
				b = strEQ1.equalsIgnoreCase(strOD);
			}
			double tmp = SystemUtil.millis() - start;
			if (fastestEqualIgnoreCaseFalse > tmp) fastestEqualIgnoreCaseFalse = tmp;
		}
		print.e("equalsIgnoreCase false:" + fastestEqualIgnoreCaseFalse);
	}`

Result:
equal true:0.3223149999976158
equal false:0.3223139997571707
equalsIgnoreCase true:133.16265799989924
equalsIgnoreCase false:147.4729599999264

equal is MUCH faster